subcommand/server-acl-init/command.go

```diff
@@ -720,11 +801,32 @@ func (c *Command) configureConnectInject(logger hclog.Logger, consulClient *api.
 			"ServiceAccountJWT": string(saSecret.Data["token"]),
 		},
 	}
+
+	// Add options for mirroring namespaces
+	if c.flagEnableSyncK8SNSMirroring {
+		authMethodTmpl.Config["MapNamespaces"] = true
+		authMethodTmpl.Config["ConsulNamespacePrefix"] = c.flagSyncK8SNSMirroringPrefix
+	}
+
+	// Set up the auth method in the specific namespace if not mirroring
+	// If namespaces and mirroring are enabled, this is not necessary because
+	// the auth method will fall back to being created in the Consul `default`
+	// namespace automatically, as is necessary for mirroring.
+	// Note: if the config changes, an auth method will be created in the
+	// correct namespace, but the old auth method will not be removed.
+	writeOptions := api.WriteOptions{}
+	if c.flagEnableNamespaces && !c.flagEnableSyncK8SNSMirroring {
+		writeOptions.Namespace = c.flagConsulSyncDestinationNamespace
```

Comment:
If the namespace doesn't exist, we need to create it. Otherwise we get these errors:
```
2020-02-05T01:11:13.727Z [ERROR] Failure: creating auth method consul-k8s-auth-method: err="Unexpected response code: 500 (Namespace "luke" does not exist)"
```

Reply:
Updated.